plugin/storage/cassandra/spanstore/operation_names.go

```diff
@@ -25,23 +26,51 @@ import (
 	"github.com/jaegertracing/jaeger/pkg/cache"
 	"github.com/jaegertracing/jaeger/pkg/cassandra"
 	casMetrics "github.com/jaegertracing/jaeger/pkg/cassandra/metrics"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
 )
 
 const (
-	insertOperationName = `INSERT INTO operation_names(service_name, operation_name) VALUES (?, ?)`
-	queryOperationNames = `SELECT operation_name FROM operation_names WHERE service_name = ?`
+	// LatestVersion latest version of operation_names table schema, increase the version if your table schema changes require code change
+	LatestVersion = 1
+	// TableQueryStmt the query statement used to check if a table exists or not
+	TableQueryStmt = "SELECT * from %s limit 1"
 )
 
+type schemaMeta struct {
+	TableName       string
+	InsertStmt      string
+	QueryByKindStmt string
+	QueryStmt       string
+}
+
+var schemas = []schemaMeta{
+	{
+		TableName:       "operation_names",
+		InsertStmt:      "INSERT INTO %s(service_name, operation_name) VALUES (?, ?)",
+		QueryByKindStmt: "SELECT operation_name FROM %s WHERE service_name = ?",
+		QueryStmt:       "SELECT operation_name FROM %s WHERE service_name = ?",
+	},
+	{
+		TableName:       "operation_names_v2",
+		InsertStmt:      "INSERT INTO %s(service_name, span_kind, operation_name) VALUES (?, ?, ?)",
+		QueryByKindStmt: "SELECT span_kind, operation_name FROM %s WHERE service_name = ? AND span_kind = ?",
+		QueryStmt:       "SELECT span_kind, operation_name FROM %s WHERE service_name = ?",
+	},
+}
+
 // OperationNamesStorage stores known operation names by service.
 type OperationNamesStorage struct {
 	// CQL statements are public so that Cassandra2 storage can override them
-	InsertStmt     string
-	QueryStmt      string
-	session        cassandra.Session
-	writeCacheTTL  time.Duration
-	metrics        *casMetrics.Table
-	operationNames cache.Cache
-	logger         *zap.Logger
+	SchemaVersion   int
+	TableName       string
+	InsertStmt      string
+	QueryStmt       string
+	QueryByKindStmt string
```

Comment:
* why do these need to be public?
* you already have the struct above containing the same fields, so you can just have a single field `table schemaMeta`